votingsystem/src/general_meeting.rs

```diff
@@ -0,0 +1,271 @@
+    }
+
+    fn end_genesis(&mut self) {}
+}
+
+
+impl TxOwner for Context {
+    fn block_opened(&mut self, _: &Header) -> Result<(), HeaderError> {
+        Ok(())
+    }
+
+    fn execute_transaction(&mut self, transaction: &Transaction) -> Result<TransactionOutcome, ()> {
+        if let Err(error) = self.excute_tx(transaction) {
+            match error {
+                ExecuteError::InvalidMetadata => Err(()),
+                ExecuteError::InvalidSign => Err(()),
+                ExecuteError::InvalidFormat => Err(()),
+                ExecuteError::InvalidAdmin => Err(()),
+            }
+        } else {
+            Ok(Default::default())
+        }
+    }
+
+    fn check_transaction(&self, transaction: &Transaction) -> Result<(), coordinator::types::ErrorCode> {
+        let todo_fixthis: coordinator::types::ErrorCode = 3;
+        assert_eq!(transaction.tx_type(), "CreateGeneralMeeting");
+        let tx: OwnTransaction = serde_cbor::from_slice(&transaction.body()).map_err(|_| todo_fixthis)?;
+        tx.verify().map_err(|_| todo_fixthis)?;
+
+        let admin_public_key = tx.signer_public;
+        if !self.storage().has(admin_public_key.as_ref()) {
+           return Err(ExecuteError::InvalidAdmin).map_err(|_|todo_fixthis);
+        }
+
+        Ok(())
+    }
+
+    fn block_closed(&mut self) -> Result<Vec<Event>, CloseBlockError> {
+        Ok(Vec::new())
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TimeStamp{
+    time: u64,
+}
+
+
+impl TimeStamp {
+    fn set_time(&mut self, time: u64){
+        self.time = time;
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct GeneralMeetingId {
+    id: u8,
+}
+
+impl GeneralMeetingId {
+    pub fn new ()->Self {
+        let mut rng = rand::thread_rng();
+        let random_id = rng.gen();
+        Self{
+            id: random_id
+        }
+    }
+}
+
+impl AsRef<[u8]> for GeneralMeetingId{
+    fn as_ref(&self)->&[u8]{
+        self.as_ref()
+    }
+}
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct VoteResult{
+    favor:u32,
+    against:u32,
+    absention: u32,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum GeneralMeetingError {
+    NoSuchMeeting,
+    MeetingExist,
+}
+
+pub struct Admin {
+    //FIXME: The public key should be hard-coded.
+    public_key: Option<Public>,
+}
+
+impl Admin {
+    pub fn new () ->Self {
+        Self{
+            public_key: None,
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct GeneralMeeting {
+    id: GeneralMeetingId,
+    number_of_agends: u32,
+    voting_end_time: TimeStamp,
+    tallying_time: TimeStamp,
+    result: Option<Vec<VoteResult>>,
+}
+
+impl GeneralMeeting {
+    pub fn new(end_time: TimeStamp, tallying_time: TimeStamp) -> Self {
+        let meeting_id= GeneralMeetingId::new();
+        let options = 4;
+        Self{
+            id: meeting_id,
+            number_of_agends: options,
```

Comment:
I used the term "agenda" as the number of votes in the meeting.
For example, if there are two agendas in the meeting like "Do you like Kimchi" and "Do you know Bibimbap", the voters should vote two times like "I don't like kimchi" and "I don't know Bibimbap".

So the number of agenda cannot be hard-coded. Someone who creates a general meeting could set the number of agendas in the meeting.